ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -49,34 +49,49 @@ func (scbuilder *SiteConfigBuilder) SetLocalExtraManifestPath(path string) {
 func (scbuilder *SiteConfigBuilder) Build(siteConfigTemp SiteConfig) (map[string][]interface{}, error) {
 	clustersCRs := make(map[string][]interface{})
 
+	err := scbuilder.validateSiteConfig(siteConfigTemp)
+	if err != nil {
+		return clustersCRs, err
+	}
+
+	for id, cluster := range siteConfigTemp.Spec.Clusters {
+		clusterCRs, err := scbuilder.getClusterCRs(id, siteConfigTemp)
+		if err != nil {
+			return clustersCRs, err
+		}
+
+		clustersCRs[siteConfigTemp.Metadata.Name+"/"+cluster.ClusterName] = clusterCRs
+	}
+
+	return clustersCRs, nil
+}
+
+func (scbuilder *SiteConfigBuilder) validateSiteConfig(siteConfigTemp SiteConfig) error {
```

Comment:
might be worthwhile adding a couple of more tests covering the error cases in the validate siteConfig call.

Reply:
there is already unit test cover all the validation case [Test_siteConfigBuildValidation ](https://github.com/openshift-kni/cnf-features-deploy/pull/864/files#diff-debded0a9a2b5f548423222493ec145e61e597d7fc5df62c4bbd3320798351c1L137)